src/main/scala/firrtl/passes/RemoveCHIRRTL.scala

```diff
@@ -118,7 +119,7 @@ object RemoveCHIRRTL extends Pass {
         set_write(rws, "wdata", "wmask")
       val mem = DefMemory(s.info, s.name, s.tpe, s.size, 1, if (s.seq) 1 else 0,
                   rds map (_.name), wrs map (_.name), rws map (_.name))
-      Block(mem +: stmts)
+      if (stmts.nonEmpty) Block(mem +: stmts) else EmptyStmt
```

Comment:
But it's still possible to write memories without ports using FIRRTL, and if that code gets compiled, other passes would break... By only changing RemoveCHIRRTL, you wouldn't be sufficiently generic? Unless I'm mistaken...